src/java.base/share/classes/jdk/internal/module/ModuleInfoWriter.java

```diff
@@ -82,86 +81,81 @@ private ModuleInfoWriter() { }
+
+                    String vs = md.rawVersion().orElse(null);
+                    if (vs != null) mb.moduleVersion(vs);
+
+                    // requires
+                    for (ModuleDescriptor.Requires r : md.requires()) {
+                        int flags = r.modifiers().stream()
+                                .map(REQUIRES_MODS_TO_FLAGS::get)
+                                .reduce(0, (x, y) -> (x | y));
+                        vs = r.rawCompiledVersion().orElse(null);
+                        mb.requires(ModuleDesc.of(r.name()), flags, vs);
+                    }
+
+                    // exports
+                    for (ModuleDescriptor.Exports e : md.exports()) {
+                        int flags = e.modifiers().stream()
+                                .map(EXPORTS_MODS_TO_FLAGS::get)
+                                .reduce(0, (x, y) -> (x | y));
+                        var targets = e.targets().stream().map(ModuleDesc::of)
+                                .toArray(ModuleDesc[]::new);
+                        mb.exports(PackageDesc.of(e.source()), flags, targets);
+                    }
+
+                    // opens
+                    for (ModuleDescriptor.Opens opens : md.opens()) {
+                        int flags = opens.modifiers().stream()
+                                .map(OPENS_MODS_TO_FLAGS::get)
+                                .reduce(0, (x, y) -> (x | y));
+                        var targets = opens.targets().stream().map(ModuleDesc::of)
+                                .toArray(ModuleDesc[]::new);
+                        mb.opens(PackageDesc.of(opens.source()), flags, targets);
+                    }
+
+                    // uses
+                    md.uses().stream().map(sn -> ClassDesc.of(sn)).forEach(mb::uses);
+
+                    // provides
+                    for (ModuleDescriptor.Provides p : md.provides()) {
+                        mb.provides(ClassDesc.of(p.service()),
+                                                 p.providers().stream()
+                                                         .map(ClassDesc::of)
+                                                         .toArray(ClassDesc[]::new));
+                    }
+                }),
+
+                // packages
+                md.packages().stream().map(PackageDesc::of).toList(),
+
+                // extra attributes
+                clb -> {
+                    // ModuleMainClass attribute
+                    md.mainClass().ifPresent(mc ->
+                            clb.with(ModuleMainClassAttribute.of(ClassDesc.of(mc))));
+
+                    // write ModuleResolution attribute if specified
+                    if (mres != null) {
+                        clb.with(ModuleResolutionAttribute.of(mres.value()));
+                    }
+
+                    // write ModuleTarget attribute if there is a target platform
+                    if (target != null && target.targetPlatform().length() > 0) {
```

Comment:
`!target.targetPlatform().isEmpty()` is probably better than a length check.

Reply:
fixed, thanks.